### ISSUE TITLE
Qwen3.5-27B-4bit VLM produces token-loop output on vision input (pre-existing on alpha)

### ISSUE THREAD

**Issue by ekryski** [ekryski/mlx-swift-lm#181]
## Symptom

\`mlx-community/Qwen3.5-27B-4bit\`, vision method, kv=none, on the standard bench dog image with prompt _"Describe what you see in this image."_ produces:

> The!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

200-token "!" repetition. Decode tok/s and TTFT look healthy (16.5 tok/s decode, 12s TTFT, 17.2 GB peak), so the model is loading and running — just emitting a pathological token loop.

## Reproduction

\`\`\`
./scripts/benchmark.sh \\
  --model mlx-community/Qwen3.5-27B-4bit \\
  --kv none \\
  --method vision \\
  --vision-prompt "Describe what you see in this image." \\
  --vision-expect "dog"
\`\`\`

Tested on M1 Max 64GB.

## Confirmed pre-existing

Reproduced on **alpha tip** (commit 7cc47e2, before any WS-C consolidation work). Surfaced during VLM smoke verification of [PR #180](https://github.com/ekryski/mlx-swift-lm/pull/180) (WS-C #8 Qwen 3.5 consolidation). The PR's typealias is shape-preserving and produces the same output as alpha — i.e., the bug is independent of the consolidation sprint.

## Suspected cause

Likely in the same family as the prefill-sync barrier issues already fixed in Gemma 3 / Gemma 4 / Mistral 3 / LFM 2 / Qwen 3 VL (see #169 for the canonical pattern):

> the VLM's prefill leaves K/V writes pending in the command buffer, so the iterator's first decode forward reads stale K/V and produces garbage logits

But Qwen 3.5 VLM is a **hybrid** model (GDN + Attention with `SSMStateCache` for the linear-attn layers), so the fix may need to also flush the SSM state writes — \`SSMStateCache.innerState()\` may need to expose the conv-state and recurrent-state arrays for the eval barrier.

## Suggested investigation steps

1. Confirm the failure is vision-input-specific — try the same prompt without an image (text-only path through the VLM).
2. Apply the standard prefill-sync barrier in \`Qwen35.prepare\` (or whatever the equivalent is on the VLM-side outer model class) and re-run the smoke. If \`SSMStateCache.innerState()\` doesn't currently expose the SSM tensors, extend it to.
3. If the barrier alone doesn't fix it, compare the first 5 logits at the channel-marker positions against the LLM (\`mlx-community/Qwen3.5-27B-4bit\` text-only) on the same prompt to localise the divergence.

## Workaround

None currently; the broken VLM is registered in \`VLMRegistry.qwen3_5_27B_4bit\` and \`VLMRegistry.qwen3_5_35B_A3B_4bit\`. Document the gap on the per-model "known gaps" table when the WS-D \`documentation/models.md\` lands.
